### ISSUE TITLE
[BUG] CocoaPods with fvm

### ISSUE THREAD

**Issue by ewilan-riviere** [fluttertools/fvm#253]
- System: macOS Big Sur
- Bug: CocoaPods with `pod` command installed but `fvm` not work with it
- Reproduce execute `fvm flutter build ios`

I'm not sure this is a bug, maybe it's my configuration but I've a problem with CocoaPods when I use `build ios` command with `fvm`

My Flutter app works with 1.22.6 and when I try to use `fvm flutter pub get` or `fvm flutter run`, I've no problem but with `fvm flutter build ios` I've this output

```
Warning: CocoaPods not installed. Skipping pod install.
  CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
  Without CocoaPods, plugins will not work on iOS or macOS.
  For more info, see https://flutter.dev/platform-plugins
To install:
  sudo gem install cocoapods

CocoaPods not installed or not in valid state.
```

But with `flutter build ios` CocoaPods wotks perfectly. I've it on my system but `fvm` seems to can't see it. Do you have any solution for this?

Thanks.

**Comment by leoafarias** [comment]
Does the issue happen every time? Also does it continue to happen even after closing an opening the terminal/ide again?

**Comment by ewilan-riviere** [comment]
Yes, I try to restart machine, restart terminal... It's not work

**Comment by leoafarias** [comment]
I will take a look. As a temporary solution. Set an alias on your bash profile.

alias f=".fvm/flutter_sdk/bin/flutter"

This will call the cached version in the project directly without pro ting through fvm.

"f build ios"

**Comment by AbirAhsan** [comment]
I solved my problem by this command
`sudo gem install -n /Users/abir/fvm/default/bin cocoapods`

**Comment by tiagocdr** [comment]
Waiting for a solution

**Comment by toebgen** [comment]
I just ran into this as well. Is there already a permanent solution to this problem?

The temporary "solution" of defining an alias helps for now.

**Comment by mbalawajdermq** [comment]
For anyone interested - I just ran into this issue and I think i found the root cause.

During execution of `flutter build` fvm looks for flutter executable location (`which flutter`) **and replaces PATH entries for found location**. 

See lib/src/utils/helpers.dart:92.

So basically, if your flutter instance was installed via homebrew, then your PATH should contain `/opt/homebrew/bin`. FVM however removes `/opt/homebrew/bin` from PATH and puts `/opt/homebrew/bin/flutter` instead. Not sure what is the reason for that.
And if your cocoapods instance were installed via homebrew too, then fvm has just removed it from PATH.

For now the best workaround I found is to install standalone flutter instance, without homebrew support.

**Comment by leoafarias** [comment]
Reopening based on the feedback from @mbalawajdermq.

Can @tiagocdr and/or @toebgen confirm that you have installed Flutter through homebrew?

**Comment by MaximeRougieux** [comment]
**TL:DR**
Hello there @leoafarias, I ran into a similar issue that I was able to reproduce in this git repository https://github.com/MaximeRougieux/fvm-issue-253-reproduction

Hope it can help to fix it.

**More context**
My use case is that I created a script to run tests in multiple packages (let's call it `runTestsInMultiplePackages`). I launch this script using the `fvm run dart runTestsInMultiplePackages.dart` and in this script there is a place where I do:
```dart
  final process = await Process.start(
    'fvm',
    isCurrentPackageFlutterAFlutterPackage
        ? ['flutter', 'test', '--no-pub', '--coverage']
        : ['dart', 'run', 'coverage:test_with_coverage'],
    workingDirectory: currentPackage.path,
  );

  await Future.wait(
    [
      stdout.addStream(process.stdout),
      stderr.addStream(process.stderr),
    ],
  );

  final exitCode = await process.exitCode;
```

Until there I don't have any issue, the process starts correctly, but here is the catch. One of my tests happens to run another process like this:

```dart
Process.runSync('fvm', ['dart', 'format', '.']);
```

There the script fails because `fvm` is not found in the PATH, because `fvm` stripped out `/opt/homebrew/bin` from the PATH environment, so it means `fvm` is actually removing itself from the PATH.

I really like the work you did, and I love using `fvm`, but I think this behavior is not ideal for use cases like mine.

**Comment by leoafarias** [comment]
@MaximeRougieux sorry for the delay on this. Essentially what fvm does is remove the dart version in the environment for the one within flutter, this is needed in cases where you want to run scripts that depend on the env variable

**Comment by claudneysessa** [comment]
Resolvi com brew install cocoapods

**Comment by MaximeRougieux** [comment]
@leoafarias Sorry I didn't see your answer and my reply is a bit late.

Correct me if I'm wrong, but:
* Placing the binary path **in front of** the `$PATH` env variable already guarantees that it will have the priority over other binary places that could be included in the `$PATH` env variable.

  _Example_: If `echo $PATH` returns `~/.fvm/versions/3.10.5:/opt/homebrew/bin`, even if both paths contain a `dart` executable, running `dart XXX` will use the executable in `~/.fvm/versions/3.10.5` because it's the first which appears in the `$PATH` env variable
* FVM does put its `dart` path first in the `$PATH` environment when running.

Providing this information, do you agree that we should get rid of this behavior?
